### PR TITLE
[infra] Use same version as apiVersion for stdlib & friends

### PR DESCRIPTION
**Pull request by martinbonnin**
Workaround for https://youtrack.jetbrains.com/issue/KT-58999/

Also bump compiletesting version

### CODE REVIEW

**Review by BoD — APPROVED** [review]
Nice 👍

**Comment by martinbonnin** [comment]
> w: The Kotlin/JS standard library has an older version (2.0.0-release-341) than the compiler (2.1.0). Such a configuration is not supported.

Ah! no `apiVersion` for JS I guess 🤔

**Comment by martinbonnin** [comment]
This has an impact to the pom files... 
```diff
$ diff -s apollo-annotations-jvm-4.1.1-20241128.114739-1.pom apollo-annotations-jvm-4.1.1-20241128.114856-2.pom
30a31,44
>   <dependencyManagement>
>     <dependencies>
>       <dependency>
>         <groupId>org.jetbrains.kotlin</groupId>
>         <artifactId>kotlin-stdlib</artifactId>
>         <version>2.0.0</version>
>       </dependency>
>       <dependency>
>         <groupId>org.jetbrains.kotlin</groupId>
>         <artifactId>kotlin-stdlib-common</artifactId>
>         <version>2.0.0</version>
>       </dependency>
>     </dependencies>
>   </dependencyManagement>
35d48
<       <version>2.0.0</version>
36a50
>       <version>unspecified</version>
41d54
<       <version>2.0.0</version>
42a56
>       <version>unspecified</version>
```

```diff
$ diff -s apollo-annotations-js-4.1.1-20241128.114739-1.pom apollo-annotations-js-4.1.1-20241128.114856-2.pom                                                   
31a32,40
>   <dependencyManagement>
>     <dependencies>
>       <dependency>
>         <groupId>org.jetbrains.kotlin</groupId>
>         <artifactId>kotlin-stdlib-common</artifactId>
>         <version>2.0.0</version>
>       </dependency>
>     </dependencies>
>   </dependencyManagement>
36c45
<       <version>2.0.0</version>
---
>       <version>2.1.0</version>
42c51
<       <version>2.1.0</version>
---
>       <version>2.0.0</version>
48d56
<       <version>2.0.0</version>
49a58
>       <version>unspecified</version>
```

I _think_ this is fine (requires != force) and it's a min version but something to keep an eye on

**Comment by martinbonnin** [comment]
For the record, before that PR, we shipped with `kotlin-stdlib-js:2.0.0` but were still able to compile the test binaries because of `kotlin-dom-api-compat` that forced 2.1 thanks to [`coreLibrariesVersion`](https://github.com/JetBrains/kotlin/blob/b3db94e7b0fa00b4d53b4e8310bb09b9564939b0/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/internal/KotlinDependenciesManagement.kt#L61)

```
 jsCompileClasspath - Compile classpath for 'js/main'.
+--- com.squareup.okio:okio:3.9.0
|    \--- com.squareup.okio:okio-js:3.9.0
|         +--- org.jetbrains.kotlin:kotlin-dom-api-compat:1.9.21 -> 2.1.0
|         |    \--- org.jetbrains.kotlin:kotlin-stdlib:2.1.0
|         |         \--- org.jetbrains.kotlin:kotlin-stdlib-js:2.1.0
|         \--- org.jetbrains.kotlin:kotlin-stdlib:1.9.21 -> 2.1.0 (*)
```

I think this is more correct now

**Comment by martinbonnin** [comment]
This happened twice:

```
 android application compiles and produces an apk FAILED
    java.lang.OutOfMemoryError: Java heap space
```

File: libraries/apollo-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -150,6 +150,8 @@ tasks.withType<Test> {
   addRelativeInput("testFiles", "testFiles")
   addRelativeInput("testProjects", "testProjects")
 
+  maxHeapSize = "1g"
```
Comment (martinbonnin) [inline libraries/apollo-gradle-plugin/build.gradle.kts:153]: By default Gradle uses 512m. I'm not sure why it "works on my machine". Using JDK17, I get this:

Screenshot 2024-11-28 at 16 19 47

The blue line being quite near the limit, I'm going to accept there some difference in CI and allocate more RAM here... 🤷

Comment (BoD) [reply libraries/apollo-gradle-plugin/build.gradle.kts:153]: Yeah that sounds reasonable. FWIW, I reproduced the OOME on my machine once today but previous and subsequent run were good ¯\_(ツ)_/¯

**Comment by martinbonnin** [comment]
Re https://github.com/apollographql/apollo-kotlin/pull/6293#issuecomment-2505948627, this is an issue for `main.kts` scripts:

```kotlin
@file:DependsOn("com.apollographql.apollo:apollo-api:4.1.1")

println(com.apollographql.apollo.api.ApolloResponse::class)
```

Output:

``` 
$ kotlin /Users/martinbonnin/Downloads/tmp/toto.main.kts
toto.main.kts:5:13: error: unresolved reference 'apollographql'.
```

**Pull request merged**